test/test_logging_macros.cpp

```diff
@@ -108,18 +108,31 @@ TEST_F(TestLoggingMacros, test_logging_expression) {
 }
 
 int g_counter = 0;
+bool g_function_called = false;
 
-bool mod3()
+bool not_divisible_by_three()
 {
+  g_function_called = true;
   return (g_counter % 3) != 0;
 }
 
 TEST_F(TestLoggingMacros, test_logging_function) {
+  // check that evaluation of a specified function does not occur if the severity is not enabled
+  g_rcutils_logging_default_severity_threshold = RCUTILS_LOG_SEVERITY_INFO;
   for (int i : {1, 2, 3, 4, 5, 6}) {
```

Comment:
do we need to make 6 calls here to confirm the function is not called a single time ? We could just do:
```c
g_counter = 1;
RCUTILS_LOG_DEBUG_FUNCTION(&not_divisible_by_three, "should not be logged");
EXPECT_EQ(0u, g_log_calls);
EXPECT_FALSE(g_function_called);
```

Reply:
yep you're right: 3dcd3c2864dad70396d576415813461c760e4c0f